docs/content/preview/releases/techadvisories/ta-28222.md

```diff
@@ -0,0 +1,132 @@
+        ```
+
+    1. Set the BNL hint to force the use of BNL.
+
+        ```sql
+        /*+ YbBatchedNL (t1 t2)*/
+        ```
+
+        This requires tracking of the query plans if they were using BNL before the upgrade.
+
+- [Enable cost-based Optimizer](/preview/best-practices-operations/ysql-yb-enable-cbo/) (CBO). This approach also requires requires running Analyze on the tables. Enabling CBO can change the query plans for various other queries.
+
+- If you upgrade to version 2024.2.5 or later, then set the following flags to enable legacy BNL mode:
+
+  - If your cluster has `yb_enable_optimizer_statistics = off`, set `yb_enable_cbo = legacy_bnl_mode`.
+  - If your cluster has `yb_enable_optimizer_statistics = on`, set `yb_enable_cbo = legacy_stats_bnl_mode`.
+
+## Details
+
+### Example
+
+This example illustrates how a query plan changes after an upgrade, leading to a potential performance regression.
+
+1. Set up the tables:
+
+    ```sql
+    drop table if exists t1, t2;
+
+    create table t1 (a int, b int);
+    insert into t1 select i, i from generate_series(1, 100) i;
+
+    create table t2 (k1 int, k2 int, k3 int, k4 int, v int, primary key (k1 asc));
+    create unique index i_t2_k2 on t2 (k2);
+    create index i_t2_k3 on t2 (k3);
+    insert into t2 select i, i, (i / 10) * 10, i % 10, i from generate_series(1, 1000000) i;
+    ```
+
+1. Default plan on v2.20.9 (with BNL enabled and CBO disabled):
+
+    ```sql
+    explain (summary off) select * from t1 join t2 on t1.a = t2.k3;
+    ```
+
+    ```output
+
+                                   QUERY PLAN
+    --------------------------------------------------------------------------
+     YB Batched Nested Loop Join  (cost=0.00..105.32 rows=5000 width=28)
+       Join Filter: (t1.a = t2.k3)
+       ->  Seq Scan on t1  (cost=0.00..100.00 rows=1000 width=8)
+       ->  Index Scan using i_t2_k3 on t2  (cost=0.00..5.20 rows=10 width=20)
+             Index Cond: (k3 = ANY (ARRAY[t1.a, $1, $2, ..., $1023]))
+    (5 rows)
+    ```
+
+1. Default plan on v2024.2 (with BNL enabled and CBO disabled):
+
+    ```sql
+    explain (summary off) select * from t1 join t2 on t1.a = t2.k3;
+    ```
+
+    ```output
+                                     QUERY PLAN
+    --------------------------------------------------------------------
+     Merge Join  (cost=299.66..379.66 rows=5000 width=28)
+       Merge Cond: (t1.a = t2.k3)
+       ->  Sort  (cost=149.83..152.33 rows=1000 width=8)
+             Sort Key: t1.a
+             ->  Seq Scan on t1  (cost=0.00..100.00 rows=1000 width=8)
+       ->  Sort  (cost=149.83..152.33 rows=1000 width=20)
+             Sort Key: t2.k3
+             ->  Seq Scan on t2  (cost=0.00..100.00 rows=1000 width=20)
+    (8 rows)
+    ```
+
+    After the upgrade, the same query chooses a merge join plan. Note that depending on the query, the BNL join may also be replaced by a hash join plan.
+
+### How to check or enable BNL is enabled
+
```

Comment:
neither of these really tell you how to check, just how to set

Reply:
Yes agree. ill rephrase it as "How to enable BNL"